### PR TITLE
vm.budgets.mlock_avail name2oid

### PR DESCRIPTION
**Pull request by mikusp**

### CODE REVIEW

**Comment by SuchAFuriousDeath** [comment]
You forgot to draw it into the tree :(

**Comment by mikusp** [comment]
On the second look I'm not sure if I've read the tree correctly 🤔 It looks in ghidra like 314 is a child of 313 but judging from other sysctls that feels wrong 😓

File: src/kernel/src/sysctl/mod.rs [hunk 1]
```diff
@@ -960,6 +973,38 @@ static VM_PS4DEV_TRCMEM_AVAIL: Oid = Oid {
     enabled: true,
 };
 
+static VM_BUDGETS: Oid = Oid {
+    parent: &VM_CHILDREN,
+    link: None, // TODO: Change to a proper value.
+    number: Sysctl::VM_BUDGETS,
+    kind: Sysctl::CTLFLAG_RD | Sysctl::CTLFLAG_WR | Sysctl::CTLTYPE_NODE,
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/sysctl/mod.rs:980]: Use Sysctl::CTLFLAG_RW instead of the first two

File: src/kernel/src/sysctl/mod.rs [hunk 2]
```diff
@@ -960,6 +973,38 @@ static VM_PS4DEV_TRCMEM_AVAIL: Oid = Oid {
...
+    arg2: 0,
+    name: "budgets",
+    handler: None,
+    fmt: "N",
+    descr: "VM budgets",
+    enabled: true,
+};
+
+static VM_BUDGETS_CHILDREN: OidList = OidList {
+    first: Some(&VM_BUDGETS_MLOCK_AVAIL), // TODO: use a proper value
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/sysctl/mod.rs:991]: Isn't it already the proper value? :thinking:

**Review by SuchAFuriousDeath — CHANGES_REQUESTED** [review]

**Review by SuchAFuriousDeath — APPROVED** [review]

**Pull request merged**